### PR TITLE
Fix with antd ssg

### LINKED ISSUE #12664: with-ant-design example not working with SSG on production
# Bug report

## Describe the bug

Currently, the with-ant-design example just works with SSG on development.
In production it throw `SyntaxError: Cannot use import statement outside a module` for antd components.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to [this repo](https://github.com/itsmetambui/next-with-antd) and clone the code. This is just a forked of the with-ant-design example with an only addition of a `getStaticProps` function.
2. `npm install` or `yarn` to install dependencies.
3. `npm run build` or `yarn build` to build the project.
4. `npm start` or `yarn start` to run the production build.
5. See error.

## Expected behavior

The with-ant-design example should also work in production.

## Screenshots

Screen Shot 2020-05-09 at 21 46 08

## System information

- OS: macOs
- Browser: Chrome
- Version of Next.js: 9.3.4
- Version of Node.js: 12.13.0

### PR DESCRIPTION
**Pull request by todortotev**
Closes #12664 
Closes #13604 

The culprit was not only changing the library directory, but also having the icons version past 4.0.6. Also, the components which used the library had to be adjusted according to the new changes.

Confirmed that it works both in production and development. Confirmed that it builds on dev and production.

Alt Text

### CODE REVIEW

File: examples/with-ant-design/package.json [hunk 1]
```diff
@@ -2,23 +2,25 @@
...
-    "react": "16.13.1",
-    "react-dom": "16.13.1"
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1"
   },
-  "license": "ISC"
+  "browser": {
+    "fs": false,
+    "path": false
+  }
```
Comment (lfades) [inline examples/with-ant-design/package.json:25]: What's this for?

File: examples/with-ant-design/package.json [hunk 2]
```diff
@@ -2,23 +2,25 @@
...
     "@next/bundle-analyzer": "^9.1.4",
-    "antd": "4.1.2",
+    "antd": "4.3.0",
     "babel-plugin-import": "1.13.0",
     "cross-env": "^7.0.2",
-    "dayjs": "1.8.24",
+    "dayjs": "1.8.28",
     "esm": "^3.2.25",
-    "next": "9.3.4",
+    "next": "9.4.4",
```
Comment (lfades) [inline examples/with-ant-design/package.json:17]:
```suggestion
    "next": "latest",
```

**Review by lfades — APPROVED** [review]
LGTM, thank you!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| buildDuration | 13.7s | 13.9s | ⚠️ +202ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.401 | 2.438 | ⚠️ +0.04 |
| / avg req/sec | 1041.23 | 1025.38 | -15.85 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.476 | 1.556 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1693.88 | 1606.35 | -87.53 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..21ef.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| index.html gzip | 926 B | 926 B | ✓ |
| link.html gzip | 935 B | 935 B | ✓ |
| withRouter.html gzip | 923 B | 923 B | ✓ |
| Overall change | 2.78 kB | 2.78 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| buildDuration | 14.6s | 14.4s | -128ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..21ef.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _error.js | 877 kB | 877 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 877 kB | 877 kB | ✓ |
| link.js | 915 kB | 915 kB | ✓ |
| routerDirect.js | 907 kB | 907 kB | ✓ |
| withRouter.js | 907 kB | 907 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| buildDuration | 14s | 13.9s | -75ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.466 | 2.439 | -0.03 |
| / avg req/sec | 1013.9 | 1024.91 | ⚠️ +11.01 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.569 | 1.609 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 1593.12 | 1553.32 | -39.8 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..21ef.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| index.html gzip | 926 B | 926 B | ✓ |
| link.html gzip | 935 B | 935 B | ✓ |
| withRouter.html gzip | 923 B | 923 B | ✓ |
| Overall change | 2.78 kB | 2.78 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| buildDuration | 14.8s | 15s | ⚠️ +118ms |
| nodeModulesSize | 65.2 MB | 65.2 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..21ef.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 57 kB | 57 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.67 kB | 5.67 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.5 kB | 52.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | TodorTotev/next.js fix-with-antd-ssg | Change |
| - | - | - | - |
| _error.js | 877 kB | 877 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 877 kB | 877 kB | ✓ |
| link.js | 915 kB | 915 kB | ✓ |
| routerDirect.js | 907 kB | 907 kB | ✓ |
| withRouter.js | 907 kB | 907 kB | ✓ |
| Overall change | 4.49 MB | 4.49 MB | ✓ |

</details>

</details>

File: examples/with-ant-design/package.json (hunk 1 above)
Comment (webbertakken) [reply examples/with-ant-design/package.json:25]: Shameless copy paste from my repo lol

**Pull request merged**